services/src/main/java/org/exoplatform/addons/gamification/search/RuleSearchConnector.java

```diff
@@ -152,6 +178,38 @@ private String buildQueryStatement(RuleFilter filter, long offset, long limit) {
     } else {
       query = query.replace("@domain_filtering@", "");
     }
+    
+    if (filter.getFilterType() != null && !FilterType.ALL.equals(filter.getFilterType())) {
+      FilterType filterType = filter.getFilterType();
+      String date = toMilliSecondsString(new Date(System.currentTimeMillis()));
```

Comment:
No need to do more than this : 
```suggestion
      String date = String.valueOf(System.currentTimeMillis());
```

Reply:
done!